### ISSUE TITLE
ZoomFactor reverts when using Ctrl+Scroll to change zoom.

### ISSUE THREAD

**Issue by gmanny** [MicrosoftEdge/WebView2Feedback#1022]
On page refresh, `ZoomFactor` property of the WPF control doesn't update while the control reverts to a zoom factor with which the navigation was started if the Ctrl+Scroll was used after navigation ended to change the zoom factor. Other times it reverts with the property change notification.

**Version**
SDK: 1.0.790-prerelease, 1.0.705.50
Runtime: Dev 88.0.705.81, Current Evergreen as well
Framework: WPF
OS: Win10

**Repro Steps**
View model:
```c#
public class BrowserTabViewModel : ObservableObject
{
        private double zoomFactor = 1.0;

        public double ZoomFactor
        {
            get => zoomFactor;
            set => SetProperty(ref zoomFactor, value);
        }

        public Uri BrowserSource => new Uri("https://example.com/");
}
```

XAML:
```xaml
        <wpf:WebView2 x:Name="webView"
                      ZoomFactor="{Binding ZoomFactor, Mode=TwoWay}"
                      Source="{Binding BrowserSource}" />
```

Steps:
1) After the page has loaded, use Ctrl+ScrollDown on the web view to decrease zoom factor. After this, the view model receives a property change to `0.9`
2) Press F5 inside the web view or trigger the reload programmatically. After the reload, visually, the web page reverts to the `1.0` zoom factor, but view model doesn't receive property change. More than that, the WebView's `ZoomFactor` property stays at `0.9` while the real zoom factor is `1.0`.
This can be confirmed by using Ctrl+ScrollDown on the web page again - the `ZoomFactor` property doesn't change. And if executing Ctrl+ScrollUp then, the page changes to 1.0 zoom factor and the view model receives the property change.

*Desirable behavior* is for the web view to retain the `0.9` zoom factor after refresh.

A *different undesirable behavior* occurs if this workaround is added:

View model:
```c#
public class BrowserTabViewModel : ObservableObject
{
        private double zoomFactor = 1.0;

        public Uri BrowserSource => new Uri("https://example.com/");

        public double ZoomFactor
        {
            get => zoomFactor;
            set => SetProperty(ref zoomFactor, value);
        }

        private void ShakeZoomFactor()
        {
            double zf = zoomFactor;
            ZoomFactor = zf + 0.001;
            ZoomFactor = zf;
        }

        public void OnNavigationCompleted(CoreWebView2NavigationCompletedEventArgs e)
        {
            ShakeZoomFactor();
        }
}
```

XAML (view model method is called through the control code behind):
```xaml
        <wpf:WebView2 x:Name="webView"
                      ZoomFactor="{Binding ZoomFactor, Mode=TwoWay}"
                      NavigationCompleted="ViewModel.OnNavigationCompleted"
                      Source="{Binding BrowserSource}" />
```

On the first reload cycle, using the steps above, the zoom factor is now restored correctly to `0.9`, then perform these steps:

3) Use Ctrl+ScrollUp to change the zoom factor to 1.0. View model correctly receives property change to `1.0`.
4) Press F5 inside the web view or trigger the reload programmatically. Now the `ZoomFactor` view model property receives a property change to `0.9` after the `NavigationStart` event was triggered. And the web view displays in zoom factor `0.9`.

This is undesirable, since the last zoom factor before reload was `1.0`. Desirable behavior is for the zoom factor to stay at 1.0 instead of the web view remembering what it was at navigation complete.

The full workaround looks like the code below, but with it, the page jitters on every reload when the zoom factor was changed by the user.

```c#
public class BrowserTabViewModel : ObservableObject
{
        private double zoomFactor = 1.0;
        private double storedZoomFactor = 1.0;

        public Uri BrowserSource => new Uri("https://example.com/");

        public double ZoomFactor
        {
            get => zoomFactor;
            set => SetProperty(ref zoomFactor, value);
        }

        private void ShakeZoomFactor()
        {
            double zf = zoomFactor;
            ZoomFactor = zf + 0.001;
            ZoomFactor = zf;
        }

        public void OnNavigationStarted(CoreWebView2NavigationStartingEventArgs e)
        {
            storedZoomFactor = zoomFactor;
        }

        public void OnNavigationCompleted(CoreWebView2NavigationCompletedEventArgs e)
        {
            if (Math.Abs(storedZoomFactor - zoomFactor) < 1e-7)
            {
                ShakeZoomFactor();
            }
            else
            {
                ZoomFactor = storedZoomFactor;
            }
        }
}
```

[AB#32047026](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/32047026)

**Comment by champnic** [comment]
Thanks for the detailed report @gmanny! We're currently making changes to how zoom works, and I'll make sure we take this into account.

**Comment by Kindles7** [comment]
Hello. The topic is almost a year old, but the bug is still present. I am writing in the hope of speeding up the solution of the problem at least minimally. 

It would be great if the zoom memory was different for each host and a full restart of WebView2 restored the last value. Thank you.

**Comment by awakecoding** [comment]
I think I'm facing the same bug in Remote Desktop Manager, and it's been driving me nuts for years already. I gave it another try today (I work for Devolutions) and managed to go a little further than the original workaround suggested. In my case, I have a mixed DPI environment with a .NET application running with system DPI awareness. The WebView2 ZoomFactor keeps being reset to 2 because I have 200% DPI scaling, even though the proper value to get the right size is a ZoomFactor of 1. We use WebView2 to present a dashboard view of a connection entry in Remote Desktop Manager, and the zoom factor just keeps changing back to 2 when selecting different connection entries. Most of the time, but not always. Anyway, here's what I found:

1) Remote Desktop Manager calls [CoreWebView2.Navigate](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.navigate) to show a different dashboard view.
2) Some internal WebView2 inconsistent, broken logic happens here that changes ZoomFactor without triggering the [ZoomFactorChanged event handler](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2controller.zoomfactorchanged)
3) The [NavigationStarting](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.navigationstarting) event handler is called, with ZoomFactor either reset or left to the original value

Now if I want to force-reset the zoom factor to the same value, it's doable by changing ZoomFactor inside the NavigationStarting event handler, but since it *sometimes* gets overwritten prior to being called, it is impossible to reliably save the previous zoom factor modified by the user with ctrl+scroll. The [state machine of navigation events *begins*](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/navigation-events) with the NavigationStarting event, so what happens in between CoreWebView2.Navigate and that first event is a mystery, and it's also where the bug seems to happen.

Could we get assistance on this? Thanks!
